opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/sitecustomize.py

```diff
@@ -112,7 +112,7 @@ def _load_configurators():
 def initialize():
     # prevents auto-instrumentation of subprocesses if code execs another python process
     environ["PYTHONPATH"] = sub(
```

Comment:
Can you also refactor the substitution into a function and add some windows and unix test cases for it?

Reply:
I have pushed the refactor. However, since it seems there are no tests for auto-instrumentation and sitecustomize, I propose creating a separate ticket to add such tests.

Reply:
Here's the ticket I created.
https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1104

Reply:
Thanks for opening the ticket, I agree we could use more tests here overall. I think it's still worthwhile to add tests for the change in this PR, e.g. if you add a test case for when the PYTHONPATH doesn't have pathsep (just a single directory in the path), it would catch the bug I mentioned in my other comment